From 95520fdd20ee6585e4b812c9d1680f09183a2ffa Mon Sep 17 00:00:00 2001
From: Thomas Betker <thomas.betker@rohde-schwarz.com>
Date: Sat, 11 May 2013 11:10:38 +0200
Subject: [PATCH 0107/1566] xilinx_emacps: Fix panic in xemacps_mii_probe()

The code checks phydev only when phy_node != NULL, resulting in a
panic when phy_node == NULL (because phydev == NULL). We now check
phydev also when phy_node == NULL, generating an error message.

This issue was introduced by the commit "xilinx_emacps: Add support for
EMIO" on Apr 03, 2013.

Signed-off-by: Thomas Betker <thomas.betker@rohde-schwarz.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/net/ethernet/xilinx/xilinx_emacps.c |    9 ++++-----
 1 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_emacps.c b/drivers/net/ethernet/xilinx/xilinx_emacps.c
index f64fca9..f33aa64 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emacps.c
+++ b/drivers/net/ethernet/xilinx/xilinx_emacps.c
@@ -802,11 +802,10 @@ static int xemacps_mii_probe(struct net_device *ndev)
 					&xemacps_adjust_link,
 					0,
 					lp->phy_interface);
-		if (!phydev) {
-			dev_err(&lp->pdev->dev, "%s: no PHY found\n",
-			ndev->name);
-			return -1;
-		}
+	}
+	if (!phydev) {
+		dev_err(&lp->pdev->dev, "%s: no PHY found\n", ndev->name);
+		return -1;
 	}
 
 	dev_dbg(&lp->pdev->dev,
-- 
1.7.5.4

